### DISCUSSION TITLE
LearnPy-eda - Kunci kode tidak sesuai pertanyaan

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#36] [Q&A] [answered]
Pada latihan Learnpy-eda dari https://gitlab.com/algoritma4students/academy-python/exercise/learnpy-da

Pada pertanyaan nomor 4: 
Compare the revenue of those 10 items between France (use France data) and all over the country (use all the data) in order to see the proportion of France market compared to all the market. What is the proportion of France market?

setelah saya [hitung](https://drive.google.com/file/d/1-fKYKNcFxF7sSnihtgE-5JYCJJxUkU7E/view?usp=sharing) berdasarkan top 10 items dari soal sebelumnya dengan:
```
store = pd.read_csv('data_input/online-retail.csv', encoding='ISO-8859-1', parse_dates=['InvoiceDate'])
store_france = store[store["Country"] == "France"]
store_france_bitna = store_france.dropna(thresh=7)
store_france_bitna["CustomerID"] = store_france_bitna["CustomerID"].fillna(store_france_bitna["CustomerID"].mode()[0])
store_france_bitna["month"] = store_france_bitna.InvoiceDate.dt.month_name()
sf = store_france_bitna.copy()
```

```
# top 10 most frequent customer
top10Cust_in_val = sf.CustomerID.value_counts(sort=True).head(10).index.values
top10Cust_in_val 
```
output = array([12681., 12682., 12567., 12637., 12683., 12731., 12536., 12700.,
       12714., 12583.])

```
#top 3 month
c1 = sf["CustomerID"].isin(top10Cust_in_val)
data_cid = sf[c1]
top3_month = data_cid.month.value_counts().head(3).index.values
```
output = array(['October', 'December', 'November'], dtype=object)

```
c2 = sf["month"].isin(top3_month)
data_cid2 = sf[c1 & c2]
top10_item = data_cid2.Description.value_counts().head(10).index.values
```
output
array(['POSTAGE', 'RABBIT NIGHT LIGHT',
       'LUNCH BOX WITH CUTLERY RETROSPOT',
       'ROUND SNACK BOXES SET OF4 WOODLAND',
       'RED TOADSTOOL LED NIGHT LIGHT', 'PLASTERS IN TIN CIRCUS PARADE',
       'LUNCH BAG RED RETROSPOT', 'RED RETROSPOT MINI CASES',
       'STRAWBERRY LUNCH BOX WITH CUTLERY',
       'PLASTERS IN TIN WOODLAND ANIMALS'], dtype=object)

```
#top10_item
sf["revenue"] = sf.Quantity * sf.UnitPrice
store["revenue"] = store.Quantity *store.UnitPrice
```

```
#top 10 rev - france
condTop10Item = sf.Description.isin(top10_item)
dataFrance = sf[condTop10Item]
top10_france = dataFrance.groupby(by="Description").sum()
top10_france.revenue
# output =
Description
LUNCH BAG RED RETROSPOT                1503.15
LUNCH BOX WITH CUTLERY RETROSPOT       1180.65
PLASTERS IN TIN CIRCUS PARADE          1840.35
PLASTERS IN TIN WOODLAND ANIMALS       1868.40
POSTAGE                               15065.00
RABBIT NIGHT LIGHT                     7275.12
RED RETROSPOT MINI CASES               1654.95
RED TOADSTOOL LED NIGHT LIGHT          2168.10
ROUND SNACK BOXES SET OF4 WOODLAND     1828.95
STRAWBERRY LUNCH BOX WITH CUTLERY       826.20
Name: revenue, dtype: float64
```

```
#top 10 rev store
condTop10Item = store.Description.isin(top10_item)
datastore = store[condTop10Item]
top10_store = datastore.groupby(by="Description").sum()
top10_store.revenue
# output
Description
LUNCH BAG RED RETROSPOT               34897.31
LUNCH BOX WITH CUTLERY RETROSPOT       7832.92
PLASTERS IN TIN CIRCUS PARADE          9961.21
PLASTERS IN TIN WOODLAND ANIMALS      13296.92
POSTAGE                               66230.64
RABBIT NIGHT LIGHT                    66756.59
RED RETROSPOT MINI CASES              12595.23
RED TOADSTOOL LED NIGHT LIGHT         23651.68
ROUND SNACK BOXES SET OF4 WOODLAND    24317.67
STRAWBERRY LUNCH BOX WITH CUTLERY      9712.88
Name: revenue, dtype: float64
```
### hasil
```
#proportion 
proportion = (top10_france.revenue.sum() / top10_store.revenue.sum()) * 100 
# hasil = 13.08 %
```

namun dengan pertanyaan 'Compare the revenue of those 10 items between France (use France data) and all over the country (use all the data) in order to see the proportion of France market compared to all the market. What is the proportion of France market?'
kunci jawabannya tidak mempergunakan top10 item tersebut, langsung begini:

```
data = pd.read_csv('data_input/online-retail.csv', encoding='ISO-8859-1', parse_dates=['InvoiceDate'])
condition_france = data['Country'] == 'France'
data['Revenue'] = data['Quantity'] * data['UnitPrice']
france = data[condition_france]
proportion = france['Revenue'].sum() / data['Revenue'].sum()

print('France Revenue: ', france['Revenue'].sum() )
print('Total Revenue : ', data['Revenue'].sum())
print('France Proportion: ', round(proportion*100, 2), '%')
```

- Maksud pertanyaan saya sebenarnya mau konfirmasi ini, apakah memang kode saya salah, atau saya salah paham memahami kode kunci soalnya atau seperti apa? Terima kasih.

**Comment by VictorNugraha** [comment, marked as answer]
Hi @Hertzspr !

Terima kasih atas antusiasmenya untuk mendalami eksplorasi data dengan mengerjakan Learnpy yang sudah kami sediakan.

Disini kami ingin mengkonfirmasi bahwa pilihan jawaban dan kunci jawaban yang kami sediakan memang ada kesalahan, mohon maaf atas kesalahan yang terjadi pada nomer tersebut. Untuk kesalahan tersebut akan segera kami perbaiki. 

Maka dari itu, untuk nomer 4 bisa dihiraukan terlebih dahulu untuk jawaban yang kami berikan dan Bapak bisa tetap yakin dengan code berserta jawaban yang sudah didapatkan, dikarenakan jika kami coba lihat dari hasil pengerjaan Bapak pada nomer tersebut, setiap proses yang Bapak lakukan sudah tepat. Good Job! 

Semoga dengan jawaban ini dapat menjawab keraguan Bapak untuk soal tersebut. Terima kasih :D

**Reply by Hertzspr** [reply]
Terima kasih.
